src/Markdig/MarkdownPipeline.cs

```diff
@@ -100,17 +100,19 @@ internal sealed class HtmlRendererCache : ObjectCache<HtmlRenderer>
 
             private readonly MarkdownPipeline _pipeline;
             private readonly bool _customWriter;
+            private readonly MarkdownParserContext? _context;
 
-            public HtmlRendererCache(MarkdownPipeline pipeline, bool customWriter = false)
+            public HtmlRendererCache(MarkdownPipeline pipeline, bool customWriter = false, MarkdownParserContext? context = null)
```

Comment:
The `MarkdownParserContext` may be different for each parsing operation and shouldn't be cached between uses of the renderer.

Instead of storing the context on `HtmlRendererCache`, set it after renting it like we [do with the writer](https://github.com/xoofx/markdig/blob/bd124f073472f4d394091da3fdea179253183f39/src/Markdig/MarkdownPipeline.cs#L87-L90) and then null it out in `Reset`.